### ISSUE TITLE
Slack message attachments is JSON encoded twice when using custom endpoint url.

### ISSUE THREAD

**Issue by avelis** [lamby/django-slack#23]
The two If checks for attachments are located here:
https://github.com/lamby/django-slack/blob/master/django_slack/api.py#L59
and
https://github.com/lamby/django-slack/blob/master/django_slack/api.py#L67

In the event that you want to send a message attachment with custom endpoint url. The payload data will in turn encode the attachments value twice.

Slack in turn ignores the attachment value and does not display the richly formatted messages.

**Comment by avelis** [comment]
@lamby I am positive the logical flow is less than ideal. Given the variables in play I found it the path of least resistance.

Open to suggestions on better providing a more readable solution while also achieving the same result.

**Comment by lamby** [comment]
Why not simply merge the two conditionals?

**Comment by avelis** [comment]
As far as I understand it.

A custom endpoint will need attachments as follows:
```python
data['attachments'] = attachments
```
This ensures richly formatted values are only encoded once because when a custom endpoint is detected the entire data dictionary is json encoded as a payload value.

In contrast a default endpoint will need a attachments as follows:
```python
data['attachments'] = json.dumps(attachments)
```
This is due to the fact that the postMessage API call does not json encode it's data but the attachments needs to be.

**Comment by lamby** [comment]
So: https://gist.githubusercontent.com/lamby/c46badb74d916ed4a6c8/raw/1da3a931e0cb64a62160ba11095ab72a46c951d7/%253Cstdin%253E ?

**Comment by avelis** [comment]
@lamby Reading your gist, it looks like it will fix the issue I am running into. If the gist change set makes it into the repo that should resolve this ticket.

**Comment by lamby** [comment]
Released to pypi as ``4.2.2``
